### PR TITLE
Store concept authors and contributors

### LINKED ISSUE #54: Author not synced
https://exercism-team.slack.com/archives/CR91YFNG3/p1623771241290900?thread_ts=1623770953.290600&cid=CR91YFNG3

A `force_sync: true` fixed this. Should we do a weekly check of everything with `force_sync: true` on in case we missed something? We might already be doing this as part of the bulk thing and I've just forgotten, but as this didn't work for this it's probably worth checking.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/v3-project-management/issues/272
Closes https://github.com/exercism/v3-beta/issues/54

### CODE REVIEW

File: app/commands/git/sync_authors_and_contributors.rb [hunk 1]
```diff
@@ -3,17 +3,39 @@ class SyncAuthorsAndContributors
     include Mandate
 
     def call
+      sync_exercise_authors_and_contributors!
+      sync_concept_authors_and_contributors!
```
Comment (ErikSchierboom) [inline app/commands/git/sync_authors_and_contributors.rb]: I briefly considered extracting this functionality to separate `SyncExerciseAuthorsAndContributors` and `SyncConceptAuthorsAndContributors` classes, but felt like it was okay to have them here. Let me know if you disagree.

File: app/commands/git/sync_concept.rb [hunk 2]
```diff
@@ -2,34 +2,52 @@ module Git
   class SyncConcept < Sync
     include Mandate
 
-    def initialize(concept)
+    def initialize(concept, force_sync: false)
```
Comment (ErikSchierboom) [inline app/commands/git/sync_concept.rb:5]: I've include the "add support for force syncing concepts" functionality in this PR, as it was quite related to the implemented functionality.

File: app/commands/git/sync_concept_exercise.rb [hunk 3]
```diff
@@ -27,8 +27,8 @@ def call
         prerequisites: find_concepts(exercise_config[:prerequisites])
       )
 
-      SyncAuthors.(exercise)
-      SyncContributors.(exercise)
+      SyncExerciseAuthors.(exercise)
+      SyncExerciseContributors.(exercise)
```
Comment (ErikSchierboom) [inline app/commands/git/sync_concept_exercise.rb:31]: Now that we have classes to sync concepts too, I had to rename the existing classes.

File: app/commands/git/sync_track.rb [hunk 4]
```diff
@@ -121,13 +121,6 @@ def sync_practice_exercises!
       end
     end
 
-    def track_needs_updating?
-      return true if force_sync
-      return false if synced_to_head?
-
-      true
-    end
-
```
Comment (ErikSchierboom) [inline app/commands/git/sync_track.rb:130]: This code was unused

File: app/jobs/sync_tracks_job.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# The goal of this job is to re-sync all tracks to guard against one of the
+# GitHub webhook calls failing, which would result in unsynced tracks
+class SyncTracksJob < ApplicationJob
```
Comment (ErikSchierboom) [inline app/jobs/sync_tracks_job.rb:3]: We didn't have a scheduled "sync all tracks" job, so I've added that here.

File: app/models/user/reputation_tokens/concept_author_token.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
+class User::ReputationTokens::ConceptAuthorToken < User::ReputationToken
+  params :authorship, :concept
+  category :authoring
+  reason :authored_concept
+  value 10
```
Comment (ErikSchierboom) [inline app/models/user/reputation_tokens/concept_author_token.rb:5]: I've made authoring a concept have half the reputation of authoring a concept exercise. Too high still?

File: app/models/user/reputation_tokens/concept_contribution_token.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
+class User::ReputationTokens::ConceptContributionToken < User::ReputationToken
+  params :contributorship, :concept
+  category :authoring
+  reason :contributed_to_concept
+  value 5
```
Comment (ErikSchierboom) [inline app/models/user/reputation_tokens/concept_contribution_token.rb:5]: I've made contributing a concept have half the reputation of authoring a concept exercise. Too high still?

**Comment by ErikSchierboom** [comment]
@iHiD Would you like me to work on the GUI-side of things, or will you or Karlo do that?

File: app/models/user/reputation_tokens/concept_author_token.rb (hunk 6 above)
Comment (iHiD) [reply app/models/user/reputation_tokens/concept_author_token.rb:5]: No, I think that's ok, especially as the people who do one will tend to do both.

File: app/commands/git/sync_authors_and_contributors.rb [hunk 8]
```diff
@@ -3,17 +3,39 @@ class SyncAuthorsAndContributors
...
+    end
+
+    private
+    def sync_exercise_authors_and_contributors!
       ::Exercise.find_each do |exercise|
         begin
-          SyncAuthors.(exercise)
+          SyncExerciseAuthors.(exercise)
+        rescue StandardError => e
+          Rails.logger.error "Error syncing exercise authors for #{exercise.track.slug}/#{exercise.slug}: #{e}"
```
Comment (iHiD) [inline app/commands/git/sync_authors_and_contributors.rb]: Should these be opened as track issues? If not in this PR, should we add a TODO here (and to the other places) for them?)

File: app/commands/git/sync_concept.rb (hunk 2 above)
Comment (iHiD) [reply app/commands/git/sync_concept.rb:5]: Great

File: app/commands/git/sync_authors_and_contributors.rb (hunk 8 above)
Comment (ErikSchierboom) [reply app/commands/git/sync_authors_and_contributors.rb]: Probably. Let me look into that

File: app/models/concept/authorship.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :concept,
+    inverse_of: :authorships,
+    foreign_key: :track_concept_id # TODO: Remove at ETL
+  belongs_to :author,
+    class_name: "User",
+    foreign_key: :user_id,
+    inverse_of: :authorships
+
+  # TODO: Remove at ETL
+  self.table_name = "concept_authorships"
```
Comment (iHiD) [inline app/models/concept/authorship.rb]: I think just name these to be correct now in the migrations as they're new tables

**Pull request merged**